crates/noirc_frontend/src/node_interner.rs

```diff
@@ -52,6 +52,12 @@ pub struct NodeInterner {
     // methods from impls to the type.
     structs: HashMap<StructId, Shared<StructType>>,
 
+    // Type Aliases map.
+    //
+    // Map type aliases to the actual type.
+    // When resolving types, check against this map to see if a type alias is defined.
+    type_aliases: HashMap<TyAliasId, Shared<TypeAliasTy>>,
```

Comment:
Do these need to be shared? I think we only mutate them once while creating them during name resolution when we can hand out a `&mut TypeAliasTy` and we don't need the shared mutability.